crates/bevy_ecs/src/lib.rs

```diff
@@ -1929,17 +1929,9 @@ mod tests {
         assert_eq!(&mapper.0, &[e1, e2, e3]);
     }
 
-    #[expect(
```

Comment:
I don't think we should be removing these dead_code expects. We're going to have to readd them when we remove the allow. Can we change them to allow's instead?